@here/harp-mapview/lib/image/MapViewImageCache.ts

```diff
@@ -83,10 +82,56 @@ export class MapViewImageCache {
         if (startLoading === true) {
             return ImageCache.instance.loadImage(imageItem);
         }
-
         return imageItem;
     }
 
+    /**
+     * Remove the image with this name from the cache.
+     *
+     * @param name - Name of the image.
+     * @returns `true` if item has been removed.
+     */
+    removeImage(name: string): boolean {
+        return this.removeImageInternal(name);
+    }
+
+    /**
+     * Remove images using the URL from the cache.
+     *
+     * @param url - URL of the image.
+     * @returns `true` if image has been removed. If multiple images are referring to the same
+     * image URL, they are all removed.
+     */
+    removeImageByUrl(url: string): boolean {
+        const names = this.m_url2Name.get(url);
+        if (names !== undefined) {
+            for (const name of [...names]) {
+                this.removeImageInternal(name);
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Remove images from the cache.
+     *
+     * @param itemFilter - Filter to identify images to remove. Should return `true` if item
+     * should be removed.
+     * @returns Number of images removed.
+     */
+    removeImages(itemFilter: (name: string, url: string) => boolean): number {
+        let numImagesRemoved = 0;
+        for (const [name, url] of [...this.m_name2Url]) {
```

Comment:
wrapping with `[... ]` doesn't seem to be needed

Reply:
Same as above